### PR TITLE
[UXPT-3056] Add props for onBeforeOpen and onBeforeClose to Popover

### PR DESCRIPTION
**Pull request by pcln-jason-v**
This PR adds the `onBeforeOpen` and `onBeforeClose` props to the Popover.

### CODE REVIEW

File: packages/popover/src/usePopover/usePopover.js [hunk 1]
```diff
@@ -27,9 +38,15 @@ function usePopover({ openOnFocus, openOnHover, openOnMount, placement, onClose,
     !open && onClose && onClose()
   }
 
-  const handleToggle = (evt) => {
+  const handleToggle = (e) => {
     setOpen((isOpen) => {
-      isOpen ? onClose?.(evt) : onOpen?.(evt)
+      if (isOpen) {
+        onBeforeClose?.()
```
Comment (DerrickTheodore) [inline packages/popover/src/usePopover/usePopover.js:44]: Hey, I may not have the context here, but why is it that  `onBeforeClose?.()` and  `onBeforeOpen?.()` do not pass the `e`?

Comment (pcln-jason-v) [reply packages/popover/src/usePopover/usePopover.js]: Hey @DerrickTheodore, thanks for checking this out.  Just added the `e` back to the `onBeforeClose` and `onBeforeOpen`.

**Review by DerrickTheodore — APPROVED** [review]
LGTM

**Review by craigpalermo — APPROVED** [review]

**Review by craigpalermo — CHANGES_REQUESTED** [review]
Just needs a `rush change`, but otherwise, LGTM.

**Review by craigpalermo — APPROVED** [review]

**Pull request merged**